Sources/Support/StoreMessagesHelper.swift

```diff
@@ -47,62 +59,47 @@ class StoreMessagesHelper {
         guard !self.showStoreMessagesAutomatically else {
             return
         }
-        // Compilation fails in older xcode versions with a similar issue to https://github.com/apple/swift/issues/57379
-        #if swift(>=5.8)
-        Task(priority: .medium) { [weak self] in
-            guard let storeMessagesProvider = self?.storeMessagesProvider else {
-                return
-            }
-            for try await message in storeMessagesProvider.messages {
-                await MainActor.run { [weak self] in
-                    self?.deferredMessages.append(message)
-                }
-            }
+
+        for try await message in self.storeMessagesProvider.messages {
+            self.deferredMessages.append(message)
         }
-        #else
-        Logger.error("Defer messages are not enabled when compiling using Xcode <14.2")
-        #endif
     }
 
     @available(iOS 16.4, *)
     @available(macOS, unavailable)
     @available(watchOS, unavailable)
     @available(tvOS, unavailable)
-    @MainActor
-    func showStoreMessages(types: Set<StoreMessageType>) {
+    func showStoreMessages(types: Set<StoreMessageType>) async {
```

Comment:
This didn't need to be `@MainActor`. Since the method is already `async`, we can `await` on the call to `display(in:)`.